### PR TITLE
Project layout and platecheck endpoint

### PR DESCRIPTION
**Pull request by indiependente**
**FEATURES**

This PR is proposing the initial design of the backend: the project is following the layout below, each will be a package exporting structs, interfaces and implementations.

 - `models`: holds all the data types in use by the business logic
 - `platechecker`: business logic
 - `store`: data store implementations for the business logic
 - `server`: exposes the business logic over HTTP (pluggable and swappable)

**Business Logic**

The `platechecker` package exposes the `PlateChecker` interface which will hold the main business logic:
 - right now the only type implementing this interface is the `PlateChecker` in the `platechecker/dummy` package.

**Store**

The `store` package exposes the `Store` interface which will persist the data:
 - right now the only type implementing this interface is the `Store` in the `store/dummy` package.

**Miscellaneous** 

 - Renamed `client` to `frontend` and `server` to `backend`
 - Using `Logger` from `github.com/indiependente/pkg/logger`
 - Using `github.com/indiependente/pkg/shutdown` to perform graceful shutdown
 - Added handler test (in the next iteration tests will be tabled)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -52,6 +52,10 @@ pre-build:
 run:
 	@cd $(SRVFOLDER) && go run main.go
 
+.PHONY: test
+test:
+	go test ./...
```
Comment (VenturaDelMonte) [inline Makefile]: come gestiamo i test? ha senso averne ora?

File: backend/server/server.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	e.Use(middleware.Logger())
+	e.Use(middleware.Gzip())
+	e.Use(middleware.RequestIDWithConfig(middleware.RequestIDConfig{
+		Generator: func() string {
+			return uuid.New().String()
+		},
+	}))
+	e.HTTPErrorHandler = customErrorHandler
+
+	return PlateCheckerServer{
```
Comment (VenturaDelMonte) [inline backend/server/server.go]: fose dovremmo usare delle variabili nella struct piu' chiare?

File: Makefile (hunk 1 above)
Comment (indiependente) [reply Makefile]: Assolutamente si, TDD!

File: backend/server/server.go (hunk 2 above)
Comment (indiependente) [reply backend/server/server.go]: Is this clear enough?
```
type PlateCheckerServer struct {
	e     *echo.Echo
	pc    platechecker.PlateChecker
	store store.PlateStore
	log   logger.Logger
}
```

File: Makefile (hunk 1 above)
Comment (indiependente) [reply Makefile]: I'll create a new PR in the next days showing how to do table tests on this handler

Comment (VenturaDelMonte) [reply Makefile]: Sounds good!

File: backend/server/server.go (hunk 2 above)
Comment (VenturaDelMonte) [reply backend/server/server.go]: Much better!

**Pull request merged**